src/components/Tool/FixEmailAddresses/FixEmailAddresses.tsx

```diff
@@ -158,33 +159,38 @@ export const FixEmailAddresses: React.FC<FixEmailAddressesProps> = ({
   const [dataState, setDataState] = useState<{
     [key: string]: PersonEmailAddresses;
   }>({});
+  const [sourceOptions, setSourceOptions] = useState<string[]>([appName]);
 
   // Create a mutable copy of the query data and store in the state
-  useEffect(
-    () =>
-      setDataState(
-        data
-          ? data.people.nodes?.reduce<{ [key: string]: PersonEmailAddresses }>(
-              (map, person) => ({
-                ...map,
-                [person.id]: {
-                  emailAddresses: person.emailAddresses.nodes.map(
-                    (emailAddress) => ({
-                      id: emailAddress.id,
-                      primary: emailAddress.primary,
-                      updatedAt: emailAddress.updatedAt,
-                      source: emailAddress.source,
-                      email: emailAddress.email,
-                    }),
-                  ),
+  useEffect(() => {
+    const existingSources = new Set<string>();
+    existingSources.add(appName);
+
+    const newDataState = data
+      ? data.people.nodes?.reduce<{ [key: string]: PersonEmailAddresses }>(
+          (map, person) => ({
+            ...map,
+            [person.id]: {
+              emailAddresses: person.emailAddresses.nodes.map(
+                (emailAddress) => {
+                  existingSources.add(emailAddress.source);
+                  return {
+                    id: emailAddress.id,
+                    primary: emailAddress.primary,
+                    updatedAt: emailAddress.updatedAt,
+                    source: emailAddress.source,
+                    email: emailAddress.email,
+                  };
```

Comment:
I guess if the email only contained these fields you could do this instead, but not required.

```suggestion
                  return emailAddress;
```